app/Http/Controllers/ProductController.php

```diff
@@ -73,7 +73,9 @@ public function update(Request $request, Product $product)
         $request->validate([
             'name' => 'required',
             'info' => 'required',
+            'product' => 'required',
         ]);
+        $product = Product::find($request->input('product'));
         $product->update($request->all());
         return redirect()->route('products.index');
     }
```

Comment:
This change is for the unittest as a test to see if it does update when I give the product in the request.